### PR TITLE
Deprecate WindowsServer1909CoreContainer AMI Family and improve error messages

### PR DESCRIPTION
**Pull request by Callisto13**
### Description

Closes https://github.com/weaveworks/eksctl/issues/3439.

A couple of things happening here:
1. Removed the `WindowsServer1909CoreContainer` AMI Family since it is now deprecated.
1. Added a fail-fast check so that if users try to use a family we don't explicitly support they will get an immediate error:
    ```
    ...
    AMI Family WindowsServer1909CoreContainer is not supported - use one of: AmazonLinux2, Ubuntu2004, Ubuntu1804, Bottlerocket, WindowsServer2019CoreContainer, WindowsServer2019FullContainer, WindowsServer2004CoreContainer
1. Improved the error message so that if another Family becomes deprecated without us knowing, we suggest users go check:
   ```
   ...
   Error: unable to determine AMI to use: error getting AMI from SSM Parameter Store: ParameterNotFound: . please verify that AMI Family is supported
   ```
1. A lot of messing about with the Validation tests, mainly putting them in an `_test` package.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [X] Manually tested
- [X] Made sure the title of the PR is a good description that can go into the release notes
- [X] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [x] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/ami/auto_resolver.go [hunk 1]
```diff
@@ -101,7 +97,7 @@ func (r *AutoResolver) Resolve(region, version, instanceType, imageFamily string
 
 	id, err := FindImage(r.api, ownerAccount, namePattern)
 	if err != nil {
-		return "", errors.Wrap(err, "error getting AMI")
+		return "", fmt.Errorf("error getting AMI from EC2 API: %w. please verify that AMI Family is supported", err)
```
Comment (aclevername) [inline pkg/ami/auto_resolver.go:100]: it has begin, RIP `errors.Wrap` 19/03/2021

**Review by aclevername — APPROVED** [review]
LGTM, unit test please :smile:

**Comment by Callisto13** [comment]
Do you need any more testz @aclevername ?

**Comment by aclevername** [comment]
> Do you need any more testz @aclevername ?

testsz look good :tada: @Callisto13

**Pull request merged**